spec file for bbswitch not compatible with kernel 5.4.x

When using a kernel 5.4.x and trying to compile (indirectly by akmod) bbswitch, we get no kernel module.
Reasons:
SUBDIRS parameter is not supported any more,
i think a simple fix in the bbswitch-kmod.spec file would be enough?

with 5.3.16 you already get a warning in the build log for bbswitch:

2020/01/03 17:57:10 akmodsbuild: + make -j8 -C /usr/src/kernels/5.3.16-300.fc31.x86_64 SUBDIRS=/tmp/akmodsbuild.0kiWMiOq/BUILD/bbswitch-kmod-0.8/_kmod_build_5.3.16-300.fc31.x86_64 modules
2020/01/03 17:57:10 akmodsbuild: make: Entering directory '/usr/src/kernels/5.3.16-300.fc31.x86_64'
2020/01/03 17:57:10 akmodsbuild: Makefile:213: ================= WARNING ================
2020/01/03 17:57:10 akmodsbuild: Makefile:214: 'SUBDIRS' will be removed after Linux 5.3
2020/01/03 17:57:10 akmodsbuild: Makefile:215:
2020/01/03 17:57:10 akmodsbuild: Makefile:216: If you are building an individual subdirectory
2020/01/03 17:57:10 akmodsbuild: Makefile:217: in the kernel tree, you can do like this:
2020/01/03 17:57:10 akmodsbuild: Makefile:218: $ make path/to/dir/you/want/to/build/
2020/01/03 17:57:10 akmodsbuild: Makefile:219: (Do not forget the trailing slash)
2020/01/03 17:57:10 akmodsbuild: Makefile:220:
2020/01/03 17:57:10 akmodsbuild: Makefile:221: If you are building an external module,
2020/01/03 17:57:10 akmodsbuild: Makefile:222: Please use 'M=' or 'KBUILD_EXTMOD' instead
2020/01/03 17:57:10 akmodsbuild: Makefile:223: ==========================================


with kernel 5.4.x you get (strange) errors like so:

2020/01/03 17:34:34 akmodsbuild: + make -j8 -C /usr/src/kernels/5.4.5-300.fc31.x86_64 SUBDIRS=/tmp/akmodsbuild.F7BntqoM/BUILD/bbswitch-kmod-0.8/_kmod_build_5.4.5-300.fc31.x86_64 modules
2020/01/03 17:34:34 akmodsbuild: make: Entering directory '/usr/src/kernels/5.4.5-300.fc31.x86_64'
2020/01/03 17:34:34 akmodsbuild: HOSTCC  scripts/basic/fixdep
2020/01/03 17:34:34 akmodsbuild: scripts/basic/fixdep.c:410:1: fatal error: opening dependency file scripts/basic/.fixdep.d: Permission denied
2020/01/03 17:34:34 akmodsbuild: 410 | }
2020/01/03 17:34:34 akmodsbuild: | ^
2020/01/03 17:34:34 akmodsbuild: compilation terminated

--- a/copr/bumblebee/bbswitch-kmod/bbswitch-kmod.spec
+++ b/copr/bumblebee/bbswitch-kmod/bbswitch-kmod.spec
@@ -3,7 +3,7 @@
 
 Name:           bbswitch-kmod
 Version:        0.8
-Release:        2%{?dist}
+Release:        3%{?dist}
 Summary:        Kernel module for powering off discrete GPU on Optimus laptops
 
 License:        GPLv2+
@@ -51,7 +51,7 @@ done
 for kernel_version in %{?kernel_versions}; do
     make %{?_smp_mflags} \
         -C "${kernel_version##*___}" \
-        SUBDIRS=${PWD}/_kmod_build_${kernel_version%%___*} \
+        M=${PWD}/_kmod_build_${kernel_version%%___*} \
         modules
 done
 
@@ -71,6 +71,10 @@ done
 
 
 %changelog
+* Tue Jan 07 2020 Andrew Gunnerson <chillermillerlong@hotmail.com> - 0.8-3
+- Change deprecated SUBDIRS= to M=
+- https://github.com/chenxiaolong/fedora-packages/issues/9
+
 * Mon May 20 2019 Andrew Gunnerson <andrewgunnerson@gmail.com> - 0.8-2
 - Rebuild against new kmodtool
 

